src/TypeSystem/Scope/ComponentScope/ComponentScope.php

```diff
@@ -43,7 +41,11 @@ public function lookupTypeFor(string $name): ?TypeInterface
         $propertyDeclarationNode = $this->componentDeclarationNode->propertyDeclarations->getPropertyDeclarationNodeOfName($name);
         if ($propertyDeclarationNode) {
             $typeReferenceNode = $propertyDeclarationNode->type;
-            return $this->resolveTypeReference($typeReferenceNode);
+            $type = $this->resolveTypeReference($typeReferenceNode);
+            if ($type instanceof EnumStaticType) {
+                $type = $type->toEnumInstanceType();
+            }
+            return $type;
```

Comment:
hmm or it could also belong inside resolveTypeReference ... i dont know... and we need unit tests for this